### PR TITLE
Oop classes and objects / it-academy.by / M-JC1-66-20 / Oxana Peregud / Done Tasks 1-4

### PR DESCRIPTION
**Pull request by OxanaPeregud**
## Pull request state

Please indicate actual status for PR:
- [x] on review
- [ ] reviewed
- [ ] fixed
- [ ] on approve
- [ ] approved

### CODE REVIEW

**Comment by OxanaPeregud** [comment]
Файлы с тестами не нужно проверять, я их для себя писала.

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package com.rakovets.course.java.core.practice.oop.classesandobjects;
+
+public class Atm {
+    private int numberBanknotes100;
+    private int numberBanknotes50;
+    private int numberBanknotes20;
+    private int amount;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java:7]: What is this?

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    public void addBanknotes20(int number) {
+        this.numberBanknotes20 = number;
+    }
+
+    public boolean isPossibleIssue(int amount) {
+        boolean isPossibleIssue = false;
+        for (int i = 0; i <= this.numberBanknotes20; i++) {
+            for (int j = 0; j <= this.numberBanknotes20; j++) {
+                for (int k = 0; k <= this.numberBanknotes20; k++) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java:32]: Wrong!
```suggestion
        for (int i = 0; i <= this.numberBanknotes20; i++) {
            for (int j = 0; j <= this.numberBanknotes50; j++) {
                for (int k = 0; k <= this.numberBanknotes100; k++) {
```

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            int option = 0;
+            for (int i = 0; i <= numberBanknotes20; i++) {
+                for (int j = 0; j <= numberBanknotes50; j++) {
+                    for (int k = 0; k <= numberBanknotes100; k++) {
+                        var b = i * 20 + j * 50 + k * 100 == amount;
+                        if (b) {
+                            String optionsCombinationBanknotes = "";
+                            optionsCombinationBanknotes = "Option " + ++option + ": ";
+                            if (i != 0)
+                                optionsCombinationBanknotes += "20 x " + i + "   ";
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Atm.java:57]: Violation of Code Convention!
Use logical block `{}`

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/AtmTest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.rakovets.course.java.core.practice.oop.classesandobjects;
+
+public class AtmTest {
+    public static void main(String[] args) {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/AtmTest.java:5]: Where are tests for `add` methods?

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Point.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.rakovets.course.java.core.practice.oop.classesandobjects;
+
+public class Point {
+    private int x;
+    private int y;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Point.java:5]: Why are you use `int`? Best `double`.

File: src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Time.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.rakovets.course.java.core.practice.oop.classesandobjects;
+
+public class Time {
+    private int hours;
+    private int minutes;
+    private int seconds;
+
+    public Time(int totalSeconds) {
+        this.seconds = totalSeconds;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/classesandobjects/Time.java:10]: Why does specification contain 2 variables for one business value?
`seconds` != `totalSeconds`

**Review by rakovets — CHANGES_REQUESTED** [review]
Good job but you need to fix following problem.

**Review by rakovets — COMMENTED** [review]
Perfect job!

**Pull request closed without merging**